cxx/src/lib/io/fileio.cc

```diff
@@ -129,4 +212,80 @@ size_t fread_from_file(TimeSeries& d,const string dir, const string dfile,
 		return ns_read;
 	}catch(...){throw;};
 }
+size_t fread_from_files(mspass::seismic::Ensemble<mspass::seismic::Seismogram> &de, const std::string dir, 
+    const std::string dfile, std::vector<long int> foffs, std::vector<long int> indexes, const long int length)
+{
+	size_t ns_read_sum;
+	int n = foffs.size();
+	de.member.resize(length);
+	FILE *fp;
+	string fname;
+	if(dir.length()>0)
+	  /* for expected context for use in python we will assume dir does not
+		have a trailing path separator so we always insert / */
+		fname = dir + "/" + dfile;
+	else
+	  /* Null as always in unix means use current directory*/
+		fname=dfile;
+	if((fp=fopen(fname.c_str(),"r")) == NULL)
+		throw MsPASSError("fread_data_from_file:  Open failed on file "+fname,ErrorSeverity::Invalid);
+	
+	for (int i = 0; i < n; ++i) {
+		size_t ns_read;
+		long int foff = foffs[i];
+		int index = indexes[i];
+		try{
+			if(foff>0)
+			{
+				if(fseek(fp,foff,SEEK_SET))
+				{
+					fclose(fp);
+				throw MsPASSError("fread_data_from_file:  fseek failure on file="+fname,ErrorSeverity::Invalid);
+				}
+			}
+			ns_read = fread((void*)de.member[index].u.get_address(0, 0), sizeof(double), 3 * de.member[index].npts(), fp);
```

Comment:
`index` should just be `i`.   Also you must trap a error condition for fread failures.   The usual test would, I think, be some version of this:
```
if(ns_read != 3*de.member[i].npts()} 
{
   de.member[i].elog.log_error( some error message);
   de.member[i].kill();
}
else
{
  de.member[i].set_live();
}
```
Noting that we also must explicitly set live all successful reads.